### PR TITLE
Fix issue for Query syntax VS facet search/folding

### PR DESCRIPTION
**Pull request by olamothe**
The expand expression for folding and facet search would not take into account the query syntax being enabled or not.

This can cause an issue with the expression being invalid when cloning the last query. So we now add the no syntax block properly around the expression to get a valid query.

Also fix an issue where the checkbox in ResultPreferences would trigger a query on initialization.

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

---
This change is [Reviewable](https://reviewable.io/reviews/coveo/search-ui/464)

### CODE REVIEW

File: src/ui/FormWidgets/Checkbox.ts [hunk 1]
```diff
@@ -71,12 +71,13 @@ export class Checkbox implements IFormWidgetWithLabel, IFormWidgetSelectable {
   }
 
   /**
-   * Checks the checkbox.
+   * Select the checkbox
+   * @param triggerChange will trigger change even if specified and not already selected
    */
-  public select() {
+  public select(triggerChange = true) {
```
Comment (francoislg) [inline src/ui/FormWidgets/Checkbox.ts]: `select(true)` and `select(false)` is not as meaningful as something like `select()` and `selectWithoutChange()`. Just consider the option, you can leave it as is :)

File: test/ui/FacetSearchParametersTest.ts [hunk 2]
```diff
@@ -22,12 +22,12 @@ export function FacetSearchParametersTest() {
...
 
-    afterEach(function () {
+    afterEach(() => {
       mockFacet = null;
     });
 
-    it('should allow to set value to search and expand it with different captions', function () {
-      var params = new FacetSearchParameters(mockFacet);
+    it('should allow to set value to search and expand it with different captions', () => {
+      let params = new FacetSearchParameters(mockFacet);
```
Comment (francoislg) [inline test/ui/FacetSearchParametersTest.ts]: Can be `const`.

File: test/ui/FacetSearchParametersTest.ts [hunk 3]
```diff
@@ -99,10 +99,10 @@ export function FacetSearchParametersTest() {
         expect(params.alwaysExclude).toContain('c');
       });
 
-      it('allows to create a group by', function () {
-        var params = new FacetSearchParameters(mockFacet);
+      it('allows to create a group by', () => {
+        const params = new FacetSearchParameters(mockFacet);
         params.setValueToSearch('qwerty');
-        var groupBy = params.getGroupByRequest();
+        let groupBy = params.getGroupByRequest();
```
Comment (francoislg) [inline test/ui/FacetSearchParametersTest.ts]: const

File: test/ui/FoldingTest.ts [hunk 4]
```diff
@@ -173,6 +173,12 @@ export function FoldingTest() {
             filterFieldRange: null
           }));
         });
+
+        it('modify query syntax', () => {
```
Comment (francoislg) [inline test/ui/FoldingTest.ts]: The other tests start with `and modify X`

File: test/ui/FacetSearchParametersTest.ts (hunk 2 above)
Comment (olamothe) [inline test/ui/FacetSearchParametersTest.ts]: It gets re-assigned further down in this test. Can't

File: test/ui/FacetSearchParametersTest.ts (hunk 3 above)
Comment (olamothe) [inline test/ui/FacetSearchParametersTest.ts]: Same.

**Pull request merged**